routing/route.cpp

```diff
@@ -354,6 +359,42 @@ void Route::Update()
   m_currentTime = 0.0;
 }
 
+void Route::AppendTraffic(Route const & route)
+{
+  CHECK(route.IsValid(), ());
+
+  if (GetTraffic().empty() && route.GetTraffic().empty())
+    return;
+
+  if (!IsValid())
+  {
+    m_traffic.assign(route.GetTraffic().cbegin(), route.GetTraffic().cend());
+    return;
+  }
+
+  // Note. At this point the last item of |m_poly| should be removed.
+  // So the size of |m_traffic| should be equal to size of |m_poly.
+  if (GetTraffic().empty())
+  {
+    CHECK_GREATER_OR_EQUAL(m_poly.GetPolyline().GetSize(), 1, ());
+    m_traffic.resize(m_poly.GetPolyline().GetSize(), SpeedGroup::Unknown);
```

Comment:
The method is called "Append", but here I see that you actually assign something to traffic. Is it OK that the m_traffic may not be empty before this line of code?